app/validation/validators.py

```diff
@@ -99,7 +99,7 @@ def __init__(self, max_decimals=0, messages=None):
 
     def __call__(self, form, field):
         data = field.raw_data[0]
-        if data is not None and '.' in data and len(data.split('.')[1]) > self.max_decimals:
+        if data and '.' in data and (self.max_decimals == 0 or len(data.split('.')[1]) > self.max_decimals):
```

Comment:
This line's pretty hard to read;

1. `if '.' in data` is equivalent to `if data and '.' in data`
1. `... and (self.max_decimals == 0 or len(data.split('.')[1]) > self.max_decimals`; looks like you could end up comparing two different types with the `>` opeartor, i.e. True > 0. This will work but it's pretty confusing.

This code should get simpler if we bring in an `INVALID_INTEGER` message but for future reference when you find yourself trying to pack this much on one line it's usually best to split it out across multiple lines and/or wrap part of it in a function.